src/Microsoft.AspNetCore.Hosting.Abstractions/IHostingEnvironment.cs

```diff
@@ -31,8 +37,15 @@ public interface IHostingEnvironment
         IFileProvider WebRootFileProvider { get; set; }
 
         /// <summary>
-        /// Gets or sets the configuration object used by hosting environment.
+        /// Gets or sets the absolute path to the directory that contains the application content files.
+        /// </summary>
+        // This must be settable!
+        string ContentRootPath { get; set; }
```

Comment:
I agree with @halter73, ContentRoot is a misleading name that could easily be confused with WebRoot.